x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdateTests.java

```diff
@@ -197,6 +198,49 @@ public void testCtor_GivenMaxNumberThreadsIsNegative() {
         assertThat(e.getMessage(), equalTo("[max_num_threads] must be a positive integer"));
     }
 
+    public void testGetUpdatedFields_GivenAll() {
+        DataFrameAnalyticsConfigUpdate update = new DataFrameAnalyticsConfigUpdate.Builder("test_job")
+            .setDescription("new description")
+            .setModelMemoryLimit(new ByteSizeValue(1024))
+            .setAllowLazyStart(true)
+            .setMaxNumThreads(8)
+            .build();
+
+        assertThat(update.getUpdatedFields(), contains("allow_lazy_start", "description", "max_num_threads", "model_memory_limit"));
```

Comment:
If `getUpdatedFields` returns `Set` then the matcher should be `containsInAnyOrder` so that it works regardless of the exact `Set` implementation.

Reply:
We want the test to fail if the set isn't sorted.